src/core/storage/compression/Compressor.h

```diff
@@ -173,27 +176,31 @@ class CompressorInternal {
 	/// Specific number of bits we re going to use for that character is determined by weight distribution
 	void initialize_trie() {
 		huff_trie *e = m_trie.data();
-		for (const auto &[key, value] : m_occurrence_symbol) {
-			e->right = nullptr;
-			e->left = nullptr;
-			e->number = value;
-			e->character = key;
-			e++;
-		}
+		for (long int *i = m_occurrence_symbol; i < m_occurrence_symbol + 256; i++)
```

Comment:
Use range-based iterators.

Reply:
Please see the new solution and tell what you think.